### ISSUE TITLE
marker.getMap Error. Second Time Marker Is Clicked Using HTMLInfoWindow

### ISSUE THREAD

**Issue by labhollow** [mapsplugin/cordova-plugin-googlemaps#2047]
**I'm submitting a ...**  (check one with "x")
- [ ] question
- [X] any problem or bug report
- [ ] feature request

**If you choose 'problem or bug report', please select OS:**  (check one with "x")
- [ ] **Android**
- [X] **iOS**

**cordova information:**  (run `$> cordova plugin list`) 

```
com.googlemaps.ios 2.5.0 "Google Maps SDK for iOS"
cordova-plugin-device 1.1.6 "Device"
cordova-plugin-googlemaps 2.2.2 "cordova-plugin-googlemaps"
cordova-plugin-ionic-webview 1.1.16 "cordova-plugin-ionic-webview"
cordova-plugin-network-information 2.0.1 "Network Information"
cordova-plugin-splashscreen 4.0.3 "Splashscreen"
cordova-plugin-statusbar 2.3.0 "StatusBar"
cordova-plugin-whitelist 1.3.2 "Whitelist"
cordova-sqlite-storage 2.2.0 "Cordova sqlite storage plugin"
ionic-plugin-keyboard 2.2.1 "Keyboard"
```

**If you use `@ionic-native/google-maps`, please tell the package.json (only `@ionic-native/core` and `@ionic-native/google-maps` are fine mostly)**

    "@ionic-native/core": "^4.5.3",
    "@ionic-native/google-maps": "^4.5.3",

**Current behavior:**
After makers are loaded, when clicking on a marker the first time, the HTMLInfoWindow displays. If you select a different marker and click it, the info window displays, but if you go back to the original marker and click it, you will get a Runtime Error: undefined is not an object (evaluating 'marker.getMap') HTMLInfoWindow.js line 202 calculate(). 

Basically you get this error the second time you click any marker.

**Expected behavior:**
HTMLInfoWindow should open normally.

**Screen capture or video record:**

**Related code, data or error log (please format your code or data):**
I have given you access to my repo here: https://github.com/labhollow/claim-explorer

Also as a note, to load the markers, do a LONG_CLICK

Giving much information, you are waiting time is less.
Thank you for your cooperation.
```

**Comment by wf9a5m75** [comment]
Open your app with Chrome Developer tools. There is the answer.
In the error stack trace log, the problem is occurred at `marker.setInfoWindowAnchor()`.

![screen shot 2018-02-04 at 8 06 57 pm](https://user-images.githubusercontent.com/167831/35787987-1c63c4a4-09e7-11e8-9ebc-c6bf5c847584.png)

And in your code, you execute `marker.setInfoWindowAnchor()` in very before `htmlInfoWindow.open()`.

```
marker.setInfoWindowAnchor(24, 0);  <---- this is problem line.
htmlInfoWindow.open(marker);
```

The error occurs when you tap on other marker during `infoWindowAnchor_changed` event is not processed. That's why `marker.getMap()` returns `undefined`.

In order to prevent this error, you need to set the `infoWindowAnchor` property when you create markers.

```
this.map.addMarker({
  icon: this.iconData,
  //disableAutoPan: true,
  position: {
    lat: lat,
    lng: lng
  },
  infoWindowAnchor: [24, 0] <-- Set the value at here
}).then((marker: Marker) => {

  marker.on(GoogleMapsEvent.MARKER_CLICK).subscribe(() => {
    // marker.setInfoWindowAnchor(24, 0);  <-- Don't change the value at here
    htmlInfoWindow.open(marker);
  });
});
```

**Comment by labhollow** [comment]
Thank you for the correction!
